toolkit/analysis_and_visualization/classification/assets/scripts/pagination.js

```diff
@@ -0,0 +1,1121 @@
+                    for (i = 1; i <= totalPage; i++) {
+                        if (i == currentPage) {
+                            html += '<li class="' + classPrefix + '-page J-paginationjs-page ' + activeClassName + '" data-num="' + i + '"><a>' + i + '<\/a><\/li>';
+                        } else {
+                            html += '<li class="' + classPrefix + '-page J-paginationjs-page" data-num="' + i + '"><a href="' + pageLink + '">' + i + '<\/a><\/li>';
+                        }
+                    }
+                    return html;
+                }
+
+                if (rangeStart <= 3) {
+                    for (i = 1; i < rangeStart; i++) {
+                        if (i == currentPage) {
+                            html += '<li class="' + classPrefix + '-page J-paginationjs-page ' + activeClassName + '" data-num="' + i + '"><a>' + i + '<\/a><\/li>';
+                        } else {
+                            html += '<li class="' + classPrefix + '-page J-paginationjs-page" data-num="' + i + '"><a href="' + pageLink + '">' + i + '<\/a><\/li>';
+                        }
+                    }
+                } else {
+                    if (attributes.showFirstOnEllipsisShow) {
+                        html += '<li class="' + classPrefix + '-page ' + classPrefix + '-first J-paginationjs-page" data-num="1"><a href="' + pageLink + '">1<\/a><\/li>';
+                    }
+                    html += '<li class="' + classPrefix + '-ellipsis ' + disableClassName + '"><a>' + ellipsisText + '<\/a><\/li>';
+                }
+
+                for (i = rangeStart; i <= rangeEnd; i++) {
+                    if (i == currentPage) {
+                        html += '<li class="' + classPrefix + '-page J-paginationjs-page ' + activeClassName + '" data-num="' + i + '"><a>' + i + '<\/a><\/li>';
+                    } else {
+                        html += '<li class="' + classPrefix + '-page J-paginationjs-page" data-num="' + i + '"><a href="' + pageLink + '">' + i + '<\/a><\/li>';
+                    }
+                }
+
+                if (rangeEnd >= totalPage - 2) {
+                    for (i = rangeEnd + 1; i <= totalPage; i++) {
+                        html += '<li class="' + classPrefix + '-page J-paginationjs-page" data-num="' + i + '"><a href="' + pageLink + '">' + i + '<\/a><\/li>';
+                    }
+                } else {
+                    html += '<li class="' + classPrefix + '-ellipsis ' + disableClassName + '"><a>' + ellipsisText + '<\/a><\/li>';
+
+                    if (attributes.showLastOnEllipsisShow) {
+                        html += '<li class="' + classPrefix + '-page ' + classPrefix + '-last J-paginationjs-page" data-num="' + totalPage + '"><a href="' + pageLink + '">' + totalPage + '<\/a><\/li>';
+                    }
+                }
+
+                return html;
+            },
+
+            // Generate HTML content from the template
+            generateHTML: function (args) {
```

Comment:
Codacy found an issue: [Method 'generateHTML' has a complexity of 27. Maximum allowed is 4.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6397592)